### PR TITLE
plugins/lsp: Add r_language_server lsp

### PR DESCRIPTION
**Pull request by BeaudanBrown**
Adds support for [languageserver](https://github.com/REditorSupport/languageserver) for R

### CODE REVIEW

File: plugins/lsp/language-servers/default.nix [hunk 1]
```diff
@@ -443,6 +443,12 @@ let
       name = "pyright";
       description = "pyright for Python";
     }
+    {
+      name = "r-language-server";
+      description = "Enable languageserver LSP for R";
```
Comment (MattSturgeon) [inline plugins/lsp/language-servers/default.nix:448]:
```suggestion
      description = "languageserver for R";
```

I think this is more consistent with other descriptions?

**Comment by MattSturgeon** [comment]
FYI, there's no need to [close a PR that has issues](https://github.com/nix-community/nixvim/pull/1630); you can push changes to the PR's branch instead (using `--force` if you need to overwrite commit history)

**Comment by BeaudanBrown** [comment]
Haha roger that, got shy

**Review by MattSturgeon — APPROVED** [review]
> Haha roger that, got shy

Glad you came back!

Looks good, thanks.

**Pull request merged**